### ISSUE TITLE
tclhash.c/734, [Tcl proc: *raw:PRIVMSG, param:  $_raw1 $_raw2 $_raw3]

### ISSUE THREAD

**Issue by ZarTek-Creole** [eggheads/eggdrop#1187]
undoubtedly linked to #1185  error on commit fb2cae85828a96af5548b47cefca498d8ddd1498
```
ash-5.1$ cat DEBUG 
Debug (eggdrop v1.9.1+stable) written Wed Aug  4 17:18:00 2021
Patch level: stable
Tcl library: /usr/lib/tcl8.6
Tcl version: 8.6.10 (header version 8.6.10)
Tcl is threaded
Compiled without IPv6 support
Compiled with TLS support
Configure flags: '--disable-ipv6' '--enable-tls' '--enable-tdns'
Compile flags: gcc -g -O2 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/usr/include -g3 -DDEBUG -DDEBUG_ASSERT -DDEBUG_MEM -DDEBUG_DNS 
Link flags: gcc -lpthread
Strip flags: touch
Context: tclhash.c/749, []
         tclhash.c/729, []
         tclhash.c/734, [Tcl proc: *raw:PRIVMSG, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/749, []
         tclhash.c/729, []
         tclhash.c/734, [Tcl proc: *raw:irc:msg, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/749, []
         tclhash.c/729, []
         tclhash.c/734, [Tcl proc: *raw:PRIVMSG, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/749, []
         tclhash.c/729, []
         tclhash.c/734, [Tcl proc: *raw:irc:msg, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/749, []
         tclhash.c/729, []
         tclhash.c/734, [Tcl proc: *raw:PRIVMSG, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/749 []

IDX ADDR                                     + PORT NICK               TYPE  INFO
--- ---------------------------------------- ------ ------------------ ----- ---------
5   0.0.0.0                                    3333 (telnet)           lstn  3333
1   0.0.0.0                                       0 -HQ                chat  flags: cptEp/0
7   91.121.173.65                            + 9000 (server)           serv  (lag: 0)
9   192.168.144.2                             34345 CREOLE.docker-egg5 bot   flags: puCosgwLiA
10  192.168.144.5                             49533 CREOLE.docker-egg3 bot   flags: puCosgwLiA
11  192.168.144.3                             46421 CREOLE.docker-egg4 bot   flags: puCosgwLiA
12  192.168.144.6                             53789 CREOLE.docker-egg2 bot   flags: puCosgwLiA
13  192.168.144.3                             50299 *                  dns   waited 0s
17  192.168.144.2                             53749 *                  dns   waited 0s
15  192.168.144.6                             57039 *                  idtw  waited 0s
19  192.168.144.6                               113 *                  idnt  (sock 15)

Not logging file main.c!
File 'language.c' accounted for 22714/22714 (ok)
File 'chanprog.c' accounted for 56/56 (ok)
File 'misc.c    ' accounted for 192176/192176 (ok)
File 'userrec.c ' accounted for 1575/1575 (ok)
File 'net.c     ' accounted for 12/12 (ok)
File 'dccutil.c ' accounted for 11728/11728 (ok)
File 'botnet.c  ' accounted for 792/792 (ok)
File 'tcl.c     ' accounted for 3328/7304 (debug follows:)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 926 :(0016) 926 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 321 :(0040) 885 :(0016) 885 :(0016) 845 :(0016)
   845 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 926 :(0016) 926 :(0016) 926 :(0016)
   926 :(0016) 926 :(0016) 926 :(0016) 926 :(0016) 926 :(0016)
   926 :(0016) 926 :(0016) 926 :(0016) 926 :(0016) 926 :(0016)
   926 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 926 :(0016) 926 :(0016) 845 :(0016)
   845 :(0016) 845 :(0016) 885 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   845 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 885 :(0016) 885 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 885 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
   885 :(0016) 845 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016) 353 :(0016)
   353 :(0016) 353 :(0016) 885 :(0016) 885 :(0016) 885 :(0016)
  
File 'tclhash.c ' accounted for 32309/32309 (ok)
File 'tclmisc.c ' accounted for 56/56 (ok)
File 'modules.c ' accounted for 1389/1389 (ok)
File 'tcldcc.c  ' accounted for 16/16 (ok)
File 'dns.c     ' accounted for 80/992 (debug follows:)
   553 :(0304) 240 :(0040) 553 :(0304) 240 :(0040) 553 :(0304)
  
File 'tls.c     ' accounted for 280/280 (ok)
Module 'uptime    ' accounted for 0/0 (ok)
Module 'console   ' accounted for 0/0 (ok)
Module 'notes     ' accounted for 0/0 (ok)
Module 'irc       ' accounted for 0/0 (ok)
Module 'ctcp      ' accounted for 0/0 (ok)
Module 'server    ' accounted for 2333/2333 (ok)
Module 'channels  ' accounted for 14695/15151 (debug follows:)
   chan.c    /47  :(01C8) chan.c    /47  :(01C8) tclchan.c /2131:(07A0)
   tclchan.c /2015:(01C8) tclchan.c /2028:(0001) tclchan.c /2033:(0020)
   tclchan.c /1998:(0001) tclchan.c /2037:(0020) tclchan.c /1998:(0001)
   tclchan.c /2041:(0020) tclchan.c /1998:(0001) tclchan.c /2131:(07A0)
   tclchan.c /2015:(01C8) tclchan.c /2028:(0001) tclchan.c /2033:(0020)
   tclchan.c /1998:(0001) tclchan.c /2037:(0020) tclchan.c /1998:(0001)
   tclchan.c /2041:(0020) tclchan.c /1998:(0001) tclchan.c /2131:(07A0)
   tclchan.c /2015:(01C8) tclchan.c /2028:(0001) tclchan.c /2033:(0020)
   tclchan.c /1998:(0001) tclchan.c /2037:(0020) tclchan.c /1998:(0001)
   tclchan.c /2041:(0020) tclchan.c /1998:(0001) tclchan.c /2131:(07A0)
   tclchan.c /2015:(01C8) irc.c     /304 :(0007) chan.c    /47  :(01C8)
   chan.c    /47  :(01C8) chan.c    /47  :(01C8) chan.c    /47  :(01C8)
   chan.c    /47  :(01C8) chan.c    /47  :(01C8) chan.c    /47  :(01C8)
   irc.c     /302 :(000C) tclchan.c /2033:(0020) tclchan.c /1998:(0001)
   tclchan.c /2037:(0020) tclchan.c /1998:(0001) tclchan.c /2041:(0020)
   irc.c     /299 :(0001) irc.c     /302 :(000E) irc.c     /447 :(0001)
   irc.c     /297 :(0020) irc.c     /304 :(0006) irc.c     /297 :(0020)
   chan.c    /47  :(01C8) chan.c    /47  :(01C8)
Module 'encryption' accounted for 0/0 (ok)
Module 'encryption2' accounted for 0/0 (ok)
Module 'eggdrop   ' accounted for 0/0 (ok)
--- End of debug memory list.
Open sockets: 5 (listen), 1, 7 (TLS), 9, 10, 11, 12, 13, 17, 15, 19 (outbuf: 00000C), 20 (file), done.
bash-5.1$ 
```
**GDB**
```
/eggdrop-data # gdb eggdrop -c core
GNU gdb (GDB) 10.2
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-alpine-linux-musl".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from eggdrop...

warning: core file may not match specified executable file.
[New LWP 82]
[New LWP 83]
Core was generated by `./eggdrop -mt /eggdrop-data/eggdrop.conf'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x000055ad1c3ec910 in dns_dcchostbyip (ip=<optimized out>, hostn=0x7f6f1f53aad4 "eggnet2.develop_lan-network", ok=1, other=<optimized out>) at dns.c:132
132             (dcc[idx].u.dns->dns_type == RES_HOSTBYIP) && (
[Current thread is 1 (LWP 82)]
(gdb) bt
#0  0x000055ad1c3ec910 in dns_dcchostbyip (ip=<optimized out>, hostn=0x7f6f1f53aad4 "eggnet2.develop_lan-network", ok=1, other=<optimized out>) at dns.c:132
#1  0x000055ad1c3ed410 in call_hostbyip (ip=ip@entry=0x7f6f1f53aabc, hostn=hostn@entry=0x7f6f1f53aad4 "eggnet2.develop_lan-network", ok=1) at dns.c:437
#2  0x000055ad1c3fc204 in sockread (s=0x7fff0261a0a0 "", len=0x7fff0261c30c, slist=0x7f6f1f668ac0, slistmax=<optimized out>, tclonly=<optimized out>) at net.c:1038
#3  0x000055ad1c3fc677 in sockgets (s=s@entry=0x7fff0261c410 "", len=len@entry=0x7fff0261c30c) at net.c:1145
#4  0x000055ad1c3f428a in mainloop (toplevel=toplevel@entry=1) at ./main.c:842
#5  0x000055ad1c3c974a in main (arg_c=3, arg_v=<optimized out>) at ./main.c:1298
(gdb) #      
```

**Comment by ZarTek-Creole** [comment]
**bt full**
```
bash-5.1# gdb eggdrop -c core
GNU gdb (GDB) 10.2
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-alpine-linux-musl".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from eggdrop...

warning: core file may not match specified executable file.
[New LWP 82]
[New LWP 83]
Core was generated by `./eggdrop -mt /eggdrop-data/eggdrop.conf'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000559ffff493dc in call_hostbyip (ip=ip@entry=0x7f684e6e084c, hostn=hostn@entry=0x7f684e6e0864 "eggnet2.develop_lan-network", ok=1) at dns.c:415
415         if ((de->lookup == RES_HOSTBYIP) && (
[Current thread is 1 (LWP 82)]
(gdb) bt full
#0  0x0000559ffff493dc in call_hostbyip (ip=ip@entry=0x7f684e6e084c, hostn=hostn@entry=0x7f684e6e0864 "eggnet2.develop_lan-network", ok=1) at dns.c:415
        de = 0x7f684e738d20
        ode = 0x7f684e738f00
        nde = 0x7f684e6dfad0
#1  0x0000559ffff58204 in sockread (s=0x7ffcae9c2050 "", len=0x7ffcae9c42bc, slist=0x7f684e6f7800, slistmax=<optimized out>, tclonly=<optimized out>) at net.c:1038
        t = {tv_sec = 1, tv_usec = 0}
        fdr = {fds_bits = {278528, 0 <repeats 15 times>}}
        fdw = {fds_bits = {0 <repeats 16 times>}}
        fde = {fds_bits = {0 <repeats 16 times>}}
        i = <optimized out>
        x = <optimized out>
        nfds_r = <optimized out>
        nfds_w = <optimized out>
        nfds_e = <optimized out>
        grab = <optimized out>
        tclsock = <optimized out>
        events = <optimized out>
        td = <optimized out>
        nfds = <optimized out>
        fd = <optimized out>
        dtn = 0x7f684e6e0840
        dtn_prev = 0x7f684e6e0980
#2  0x0000559ffff58677 in sockgets (s=s@entry=0x7ffcae9c43c0 "", len=len@entry=0x7ffcae9c42bc) at net.c:1145
        xx = "\000docker-egg5!CREOLE@docker-egg5.sur.creole.org PRIVMSG #EPiSTAXiS-DEV :1-> CREOLE.docker-egg2 | /eggdrop-data/data/secrets/CREOLE.docker-egg2.sct\000\n\000RC\000s too bouncy\r\n\000g NOTICE docker-egg1 :*** Redirect"...
        p = <optimized out>
        px = <optimized out>
        p2 = <optimized out>
        ret = <optimized out>
        i = -1365499824
        data = 0
        len2 = <optimized out>
#3  0x0000559ffff5028a in mainloop (toplevel=toplevel@entry=1) at ./main.c:842
        socket_cleanup = 0
        xx = <optimized out>
        i = 0
        eggbusy = 1
        tclbusy = 0
        buf = "\000docker-egg5!CREOLE@docker-egg5.sur.creole.org\000PRIVMSG\000#EPiSTAXiS-DEV :1-> CREOLE.docker-egg2 | /eggdrop-data/data/secrets/CREOLE.docker-egg2.sct\000 IRC\000n\000Serv REGISTER password\000,PRIVMSG:4,SAJOIN:,SAPAR"...
#4  0x0000559ffff2574a in main (arg_c=3, arg_v=<optimized out>) at ./main.c:1298
        i = <optimized out>
        xx = <optimized out>
        s = "Wed Aug  4 2021\000:15 2021"
        f = <optimized out>
        sv = {__sa_handler = {sa_handler = 0x559ffff4fd30 <got_alarm>, sa_sigaction = 0x559ffff4fd30 <got_alarm>}, sa_mask = {__bits = {0, 3017000483, 140085976788665, 202467474075899592, 
              140723237972312, 0, 140085977040544, 140085977041216, 140085977041736, 0, 140085976799869, 0, 140723237972568, 140723237972312, 3, 140085977041216}}, sa_flags = 0, 
          sa_restorer = 0x8683fbf8}
        chan = <optimized out>
        cdlim = {rlim_cur = 18446744073709551615, rlim_max = 18446744073709551615}
(gdb) 
```

**Comment by michaelortmann** [comment]
thanks for the report.
sadly the 2 stack reports do not match.
its unclear, if this is a tdns bug.
lets see if the bug is reproduceable without tdns (and without dns mod loaded)

**Comment by vanosg** [comment]
@MalaGaM please confirm this bug is fixed for you by #1186 and close if it is resolved. Thanks!

**Comment by ZarTek-Creole** [comment]
@vanosg confirmed. Thanks!
